src/infrastructure/dataBases/nodeTsKeleton/User.model.ts

```diff
@@ -41,4 +43,37 @@ export class UserModel implements IUserModel {
       return resolve(user as User);
     });
   }
+
+  private removeSessionLogout(sessionId: string) {
+    const index = dbData.logoff.findIndex((session) =>
+      BooleanUtil.areEqual(session.sessionId, sessionId),
+    );
+    if (index > -1) {
+      dbData.logoff.splice(index, 1);
+    }
+  }
+
+  async registerLogout(session: ISession): Promise<Boolean> {
+    return new Promise((resolve) => {
+      const { exp, sessionId } = session;
+      const expireAt = DateTimeUtils.toDateTMZ(exp);
+      const timeout = expireAt.getTime() - new Date().getTime();
+      setTimeout(() => this.removeSessionLogout(sessionId), timeout);
+      dbData.logoff.push({ expireAt, sessionId } as any);
+      return resolve(true);
+    });
+  }
+
+  async getBySessionId(sessionId: string): Promise<ISession> {
+    return new Promise((resolve, reject) => {
+      const founded = dbData.logoff.find((session) =>
+        BooleanUtil.areEqual(session.sessionId, sessionId),
+      );
+      if (!founded) {
+        return reject(null);
+      }
+      const domainSession: ISession = { ...founded } as any;
```

Comment:
Please try to don't use any.  If you are using TypeScript, the idea is that you control your typing.

Reply:
In this case the object does not correspond to the interface, I put @ts-ignore